src/wrapper/jnienv.rs

```diff
@@ -238,9 +276,19 @@ impl<'local> JNIEnv<'local> {
         }
     }
 
+    // FIXME: the `JNIVersion` type is a bit of a mess and should
+    // probably be reworked. In the meantime we have this alternative
+    // that just returns the int JNI version
+    fn version(&self) -> i32 {
```

Comment:
Right, I'd been thinking of just having a wrapper for a `u32` that could have `.major()` and `.minor()` getters that would just pull the upper or lower 16 bits (or conservatively could pull the lower 8 bits and the 3rd byte, while 255 is more than enough head room for now and that avoids assuming nothing weird will be done with the other bits)

Reply:
For reference here:

I updated the `JNIVersion` type to use more conventional JNI version designations (so `V1`, `V2`, `V4`, `V6` and `V8` are now `V1_1`, `V1_2`, `V1_4`, `V1_6` and `V1_8`) and the type is now a transparent `i32` wrapper that just provides a more ergonomic way to check a raw JNI version. There is no 'invalid' version any more so even if `JNIVersion` has not been updated to know about some future `V30` it is still possible to check the `.major()` and `.minor()` component and it is valid to do `Ord`ered comparisons between versions.

The public `JNIEnv::get_version()` API was renamed to `JNIEnv::version()` since that's more idiomatic for a Rust API.